buildSrc/src/main/java/org/elasticsearch/gradle/Version.java

```diff
@@ -139,13 +147,27 @@ public boolean equals(Object o) {
                 revision == version.revision &&
                 id == version.id &&
                 snapshot == version.snapshot &&
-                Objects.equals(suffix, version.suffix);
+                Objects.equals(qualifier, version.qualifier);
     }
 
     @Override
     public int hashCode() {
 
-        return Objects.hash(major, minor, revision, id, snapshot, suffix);
+        return Objects.hash(major, minor, revision, id, snapshot, qualifier);
+    }
+
+    public boolean isSameVersionNumber(Version other) {
+        return major == other.major &&
+            minor == other.minor &&
+            revision == other.revision;
+    }
+
+    public Version dropQualifier() {
```

Comment:
nit: can we use the "without" terminology? I think this better matches other code like the builders for java time stuff having eg`withTimeZone`. Drop implies mutating the current object.